servicetalk-grpc-netty/src/test/java/io/servicetalk/grpc/protoc/GeneratorTest.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright © 2019 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.grpc.protoc;
+
+import io.servicetalk.grpc.netty.TesterProto.Tester;
+
+import org.junit.Test;
+
+import java.util.List;
+
+import static io.servicetalk.grpc.protoc.Words.Blocking;
+import static io.servicetalk.grpc.protoc.Words.Rpc;
+import static java.util.Arrays.stream;
+import static java.util.stream.Collectors.toList;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+public class GeneratorTest {
+
+    @Test
+    public void testGeneratedFunctionalInterfaces() {
+
+        final List<Class<?>> generatedRpcInterfaces = stream(Tester.class.getDeclaredClasses())
+                .filter(declaredClass -> declaredClass.getAnnotation(FunctionalInterface.class) != null
+                        && !declaredClass.getSimpleName().startsWith(Blocking)
+                        && declaredClass.getSimpleName().endsWith(Rpc))
+                .collect(toList());
+
+        final List<Class<?>> generatedBlockingRpcInterfaces = stream(Tester.class.getDeclaredClasses())
+                .filter(declaredClass -> declaredClass.getAnnotation(FunctionalInterface.class) != null
+                        && declaredClass.getSimpleName().startsWith(Blocking)
+                        && declaredClass.getSimpleName().endsWith(Rpc))
+                .collect(toList());
+
+        assertThat(generatedRpcInterfaces.size(), greaterThan(0));
```

Comment:
Because we want to test all 4 different API variants and we know that `Tester` class defines them all, I would make sure that the size of both `List`s is 4.